Source/WebCore/workers/service/ServiceWorkerRegistration.cpp

```diff
@@ -311,7 +311,21 @@ void ServiceWorkerRegistration::showNotification(ScriptExecutionContext& context
         return;
     }
 
-    if (auto* serviceWorkerGlobalScope = dynamicDowncast<ServiceWorkerGlobalScope>(context)) {
+    auto* serviceWorkerGlobalScope = dynamicDowncast<ServiceWorkerGlobalScope>(context);
+
+    // If we're handling a PushNotificationEvent, this Notification will override the proposed notification
+    // instead of being shown directly.
+#if ENABLE(DECLARATIVE_WEB_PUSH)
+    if (serviceWorkerGlobalScope) {
```

Comment:
You could probably move the `#if ENABLE(DECLARATIVE_WEB_PUSH)` inside the `if(serviceWorkerGlobalScope)`, then you could delete line 328.